tests/unit/tests_high_five_base.cpp

```diff
@@ -651,25 +651,21 @@ void readWriteAttributeVectorTest() {
         Group g = file.createGroup("dummy_group");
 
         // check that no attributes are there
-        std::size_t n = g.getNumberAttributes();
-        BOOST_CHECK_EQUAL(n, 0);
+        BOOST_CHECK_EQUAL(g.getNumberAttributes(), 0);
```

Comment:
BOOST_CHECK_EQUAL, on error, this works pretty bad when trying to output a meaningful message.
I'd suggest to leave as it was (maybe changing the type to auto)